packages/x-date-pickers/src/CalendarPicker/DayPicker.tsx

```diff
@@ -43,6 +43,13 @@ export interface ExportedDayPickerProps<TDate>
    * @default () => "..."
    */
   renderLoading?: () => React.ReactNode;
+  /**
+   * Format the day name displayed in the calendar header
+   * @param {string} day The day name provided by the adapter's method `getWeekdays`
+   * @returns {string} The name to display
+   * @default (day) => day.charAt(0).toUpperCase()
+   */
+  headerDayFormatter?: (day: string) => string;
```

Comment:
The name of the days are rendered in the header but we may allow users to change this or even repeat them in the footer too, so I would avoid "header". Other props that accept function have the "get" prefix.

```suggestion
  getDayOfWeekFormatter?: (day: string) => string;
```

Reply:
I applied the modification but used `dayOfWeekFormatter` because the idea that this prop is a function is already implied by the `Formatter` suffix. Like we say `valueFormatter` and not `getValueFormatter` in the data grid